mlos_core/mlos_core/tests/optimizers/optimizer_test.py

```diff
@@ -388,14 +384,12 @@ def objective(point: pd.DataFrame) -> pd.DataFrame:
         # Test registering the suggested configuration with a score.
         observation = objective(suggestion)
         assert isinstance(observation, pd.DataFrame)
-        optimizer.register(suggestion, observation)
+        optimizer.register(suggestion, observation, context)
 
     (best_config, best_score, best_context) = optimizer.get_best_observations()
     assert isinstance(best_config, pd.DataFrame)
     assert isinstance(best_score, pd.DataFrame)
-    assert best_context is None
 
     (all_configs, all_scores, all_contexts) = optimizer.get_observations()
     assert isinstance(all_configs, pd.DataFrame)
     assert isinstance(all_scores, pd.DataFrame)
-    assert all_contexts is None
```

Comment:
Can we assert that it *is* something expected now instead of just removing the check?

Reply:
No, because only SMAC uses context, where there other optimizers do not

Reply:
Then check for that too.

Reply:
I made the associated changes